run-make: add some basic docs about the test suite's setup

jieyouxu opened the pull request:
r? @Kobzol

jieyouxu committed:
run-make: add some basic docs about the test suite's setup

Kobzol commented on tests/run-make/README.md:
````diff
@@ -0,0 +1,35 @@
+   structure
+
+   ```
+   test_specific_build_dir/
````
I would mention that this directory is inside `build/<target>/test`, so that people can find the directory and take a look at what's inside when a test fails.

jieyouxu replied:
Added the `build/<target>/test/run-make/` path.

lqd commented on tests/run-make/README.md:
```diff
@@ -0,0 +1,35 @@
+# The `run-make` test suite
+
+The `run-make` test suite contain tests which are the most flexible out of all
```
```suggestion
The `run-make` test suite contains tests which are the most flexible out of all
```

jieyouxu replied:
Fixed

jieyouxu commented:
@rustbot review

Kobzol commented on tests/run-make/README.md:
```diff
@@ -0,0 +1,35 @@
+   support library is made available as an [*extern prelude*][extern_prelude].
+3. Finally, we run the recipe binary and set `rmake_out/` as the CWD.
+
+[`run_make_support`]: https://github.com/rust-lang/rust/tree/master/src/tools/run-make-support
```
I would use a relative link to `../../src/tools/run-make-support` here. It then works both locally and on GitHub.

jieyouxu replied:
Adjusted to use relative link.

Kobzol commented on tests/run-make/README.md:
```diff
@@ -0,0 +1,35 @@
+
+## Infrastructure
+
+The implementation for collecting and building the `rmake.rs` recipes (or
```
Maybe one sentence that describes that there are two types of runmake tests would be useful.

jieyouxu replied:
Explained that there are two kinds of run-make tests.

Kobzol reviewed (commented):
Looks good, at least now there is some basic documentation about this, for the uninitiated. I left two nits, but otherwise LGTM. This should wait for https://github.com/rust-lang/rust/pull/126097, ofc.

Kobzol commented on tests/run-make/README.md:
```diff
@@ -0,0 +1,42 @@
+
+The implementation for collecting and building the `rmake.rs` recipes (or
+`Makefile`s) are in
+[`src/tools/compiletest/src/runtest.rs`](https://github.com/rust-lang/rust/blob/master/src/tools/compiletest/src/runtest.rs),
```
This link could also be relative.

jieyouxu replied:
Changed link to be relative.

Kobzol commented on tests/run-make/README.md:
```diff
@@ -0,0 +1,42 @@
+
+   and copy non-`rmake.rs` input support files over to `rmake_out/`. The
+   support library is made available as an [*extern prelude*][extern_prelude].
+3. Finally, we run the recipe binary and set `rmake_out/` as the CWD.
```
```suggestion
3. Finally, we run the recipe binary and set `rmake_out/` as the working directory.
```
Just to make it a bit clearer :)

jieyouxu replied:
Avoided using CWD abbreviation.

Kobzol reviewed (approved):
Feel free to r=me once https://github.com/rust-lang/rust/pull/126097 gets merged :)

Kobzol commented:
https://github.com/rust-lang/rust/pull/126097 has been merged.

jieyouxu commented:
Addressed review comments. I plan to add some more docs on best practices for rmake.rs tests and a high-level overview of the support library in follow-up PR(s). (Still gathering thoughts / sitting on some ideas for those.)

@bors r=@Kobzol

bors commented:
:pushpin: Commit e7409d2130b3065169ab3933fb7fb515ef0303a6 has been approved by `Kobzol`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Kobzol e7409d2130b3065169ab3933fb7fb515ef0303a6 -->
<!-- homu: {"type":"Approved","sha":"e7409d2130b3065169ab3933fb7fb515ef0303a6","approver":"Kobzol","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

jieyouxu commented:
@bors rollup

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/9433605796/job/25986334796)
Approved by: Kobzol
Pushing 13423befc40fffe23ccc6dd06868142cff9428fe to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"Kobzol","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/9433605796/job/25986334796"},"merge_sha":"13423befc40fffe23ccc6dd06868142cff9428fe"} -->

rust-timer commented:
Finished benchmarking commit (13423befc40fffe23ccc6dd06868142cff9428fe): [comparison URL](https://perf.rust-lang.org/compare.html?start=a595f3218e1204bad6f4d3451d318b2c0c56d570&end=13423befc40fffe23ccc6dd06868142cff9428fe&stat=instructions:u).

### Overall result: no relevant changes - no action needed


@rustbot label: -perf-regression



### [Instruction count](https://perf.rust-lang.org/compare.html?start=a595f3218e1204bad6f4d3451d318b2c0c56d570&end=13423befc40fffe23ccc6dd06868142cff9428fe&stat=instructions:u)
This benchmark run did not return any relevant results for this metric.

### [Max RSS (memory usage)](https://perf.rust-lang.org/compare.html?start=a595f3218e1204bad6f4d3451d318b2c0c56d570&end=13423befc40fffe23ccc6dd06868142cff9428fe&stat=max-rss)
This benchmark run did not return any relevant results for this metric.

### [Cycles](https://perf.rust-lang.org/compare.html?start=a595f3218e1204bad6f4d3451d318b2c0c56d570&end=13423befc40fffe23ccc6dd06868142cff9428fe&stat=cycles:u)
<details>
<summary>Results (primary -0.4%, secondary 2.3%)</summary>

This is a less reliable metric that may be of interest but was not used to determine the overall result at the top of this comment.

|                                    | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | 2.1%  | [2.1%, 2.1%]   | 1     |
| Regressions ❌ <br /> (secondary)  | 2.3%  | [2.3%, 2.3%]   | 1     |
| Improvements ✅ <br /> (primary)   | -3.0% | [-3.0%, -3.0%] | 1     |
| Improvements ✅ <br /> (secondary) | -     | -              | 0     |
| All ❌✅ (primary)                 | -0.4% | [-3.0%, 2.1%]  | 2     |
</details>

### [Binary size](https://perf.rust-lang.org/compare.html?start=a595f3218e1204bad6f4d3451d318b2c0c56d570&end=13423befc40fffe23ccc6dd06868142cff9428fe&stat=size:linked_artifact)
This benchmark run did not return any relevant results for this metric.

**Bootstrap**: missing data
**Artifact size**: 319.87 MiB -> 319.75 MiB (-0.04%)
